src/up.cmd.js

```diff
@@ -179,10 +179,11 @@ class UpCommand extends BuildCommand {
 
   async setup() {
     await super.init();
-    // check for git repository
-    if (!await fse.pathExists(path.join(this.directory, '.git'))) {
-      throw Error('hlx up needs local git repository.');
+    if (!this._gitHead) {
+      this._gitDir = await GitUtils.getGitDir(this.directory);
+      this._gitHead = path.join(this._gitDir, 'HEAD');
```

Comment:
why not `path.resolve()` ?

Reply:
no real reason, `this.directory` is already resolved and `getGitDir()` returns a resolved path - do you always prefer `path.resolve()`?